central/metrics/aggregator/common/tracker_config.go

```diff
@@ -0,0 +1,165 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stackrox/rox/central/metrics"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+// TrackerConfig wraps various pieces of configuration required for tracking
+// various metrics.
+type TrackerConfig[Finding Countable] struct {
+	category    string
+	description string
+	labelOrder  map[Label]int
+	getters     map[Label]func(Finding) string
+	generator   FindingGenerator[Finding]
+	gauge       func(string, prometheus.Labels, int)
+
+	// metricsConfig can be changed with an API call.
+	metricsConfig    MetricsConfiguration
+	metricsConfigMux sync.RWMutex
```

Comment:
Using RWMutex makes sense only in some very specific scenarios. Why do you think it makes sense here?